Signal/src/contact/OWSContactsManager.m

```diff
@@ -245,16 +281,23 @@ - (NSString *)unknownContactName
                              @"Displayed if for some reason we can't determine a contacts phone number *or* name");
 }
 
-- (NSString * _Nonnull)displayNameForPhoneIdentifier:(NSString * _Nullable)identifier {
-    if (!identifier) {
+- (NSString *_Nonnull)displayNameForPhoneIdentifier:(NSString *_Nullable)recipientId
+{
```

Comment:
I just rewrote this entire PR.  

* It'll now be transaction-safe; we no longer use transaction inside the "get name" methods.
* Results will now be consistent between "unattributed lookup by signal account" and "unattributed lookup by recipient id".
* Results will now be consistent between "attributed lookup by signal account" and "attributed lookup by recipient id".
* Results will now be more consistent between attributed and non-attributed names.
* There are small improvements around formatting of spaces, parentheses, etc.